lib/src/main/java/com/auth0/jwt/JWTCreator.java

```diff
@@ -19,6 +19,8 @@
  * from a given Header and Payload content.
  * <p>
  * This class is thread-safe.
+ *
+ * TODO Poovam - Should we claim thread safety since KeyProvider implementation can cause signing differ
```

Comment:
Do you mean that an implementation of KeyProvider may not be thread-safe, rendering this class not thread-safe? If so, we can claim thread safety of this class, and also document that implementations of KeyProvider must be thread-safe.